### DISCUSSION TITLE
Create Server from Backup, Add .pelicanconfig to backups

### DISCUSSION THREAD

**Discussion by notAreYouScared** [pelican-dev/panel#433] [Ideas]
Allow creating a server from a backup, Also add `.pelicanconfig` file to the backup that would include things like allocations, egg name + uuid, egg variables + values, allocation limits for resources and features, along with startup and current docker image in use.

**Comment by Simsz** [comment]
Was going to make a feature request but this appears to be what I'd want.

Here is what I had typed up:

Currently, when a game server is destroyed, there's no built-in way to restore it from backups while maintaining the original port configuration (or create a new server from backup at all).

## Proposed Solution

### Option 1: Better Backup Restoration
Allow users to restore game servers from local backups even after the original server instance has been destroyed, maintaining the original port configuration and all other game server settings (with something like a config file included in the backup).

### Option 2: Flexible Port Management During Suspension
Add an option to "un-reserve" port allocations when suspending servers, allowing those ports to be used by other server instances.

**Comment by PatrykPatryk5** [comment]
Great ideas! However, I would suggest adding an option where backups from Wings 1 could be stored on Wings 2 and vice versa. This way, if Wings 1 goes down, the servers can still be restored without issue!  

Additionally, there should be an option to designate a specific Wings instance (e.g., Wings 10) as a dedicated backup storage location where all backups are stored centrally.  

Another useful feature would be ensuring that each backup is stored in at least two locations: one on the same Wings instance as the original server and another on a different Wings instance. This would provide extra redundancy and improve reliability in case of failures.

**Reply by rmartinoscar** [reply]
I feel like using s3 with the same bucket for both nodes would be the best move for this kind of usage.
>For 3-2-1 backup its up to you to do whats required.

**Reply by PatrykPatryk5** [reply]
Using S3 with the same bucket for both nodes is a good solution, but not everyone has access to S3. That’s why it would be useful to have an option for Wings nodes to handle backups directly between each other. This way, users who don’t use S3 can still benefit from automatic redundancy.

As for 3-2-1 backups, I understand that it’s up to me to configure them, but having built-in support for multi-node backup synchronization in Wings would make the process much easier and more accessible.

**Comment by orcist** [comment]
interested in this feature, would allow shutting down servers during downtime to save on hosting cost..

**Comment by zzjrzz** [comment]
+1 it will be nice to be able to restore a server from a backup .tar.gz instead of re-creating the server and placing the files in with sftp.

**Comment by OXipro** [comment]
+1 Wil be nice to have a restore from backup or a clone server feature
